src/hooks/useCloseOnOutSideClick.ts

```diff
@@ -0,0 +1,13 @@
+import React, { useEffect } from "react";
+
+export default function (ref: React.MutableRefObject<any>, closeFunction: () => void) {
```

Comment:
Just a small nit-pick: this hook is about clicking outside. It doesn't have to know about closign. What do you think?